tests/tripal_chado/fields/ChadoFieldGetValuesListTest.php

```diff
@@ -0,0 +1,442 @@
+
+              // We assume it's not a foreign key...
+              $rel = 'base';
+              // and then check the schema to see if we're wrong :-)
+              foreach ($base_schema['foreign keys'] as $schema_info) {
+                if (isset($schema_info['columns'][ $field_column ])) {
+                  $rel = 'foreign key';
+                  $rel_table = $schema_info['table'];
+                }
+              }
+            }
+          }
+
+          // Store all the info about bundle, field, instance, schema for use in the test.
+          $info = array(
+            'field_name' => $field_name,
+            'bundle_name' => $bundle_name,
+            'bundle' => $bundle,
+            'bundle_base_table' => $bundle_base_table,
+            'base_schema' => $base_schema,
+            'field_info' => $field_info,
+            'instance_info' => $instance_info,
+            'fk_table' => $rel_table,
+          );
+
+          // Create a unique key.
+          $key = $bundle_name . '--' . $field_name;
+
+          // If this bundle uses chado and we know the fields relationship to the base
+          // chado table, then we want to index the field list by that relationship.
+          if ($rel) {
+            $this->field_list[$storage][$rel][$key] = array(
+              $field_name,
+              $bundle_name,
+              $info
+            );
+          }
+          else {
+            $this->field_list[$storage][$key] = array(
+              $field_name,
+              $bundle_name,
+              $info
+            );
+          }
+
+        }
+      }
+    }
+
+    return $this->field_list;
+  }
+
+  /**
+   * Generate fake data for a given bundle.
+   *
+   * If only the first parameter is provided this function adds fake data to the indicated
+   * chado table. If the third parameter is provided the generated fake data will
+   * have a fixed value for the indicated column.
+   *
+   * @return
+   *   Returns FALSE if it was unable to create fake data.
+   */
+  function generateFakeData($chado_table, $schema, $fixed_column = FALSE, $fk_table = FALSE) {
+
+    // First, do we have a factory? We can't generate data without one...
+    if (!Factory::exists('chado.'.$chado_table)) {
+      return FALSE;
+    }
+
+    // Create fake data -TripalTestSuite will use faker for all values.
+    if ($fixed_column === FALSE) {
+      factory('chado.'.$chado_table, 50)->create();
+      return TRUE;
+    }
+
+
+    // Attempt to create a fixed fake value.
+    // This needs to match the column type in the chado table and if the column is a
+    // foreign key, this value should match a fake record in the related table.
+    $fake_value = NULL;
+
+    // If we weren't told the related table then we assume this is a simple column (not a foreign key).
+    if ($fk_table === FALSE) {
+      $column_type = $schema[$fixed_column]['type'];
+      if (($column_type == 'int')) {
+        $fake_value = $faker->randomNumber();
```

Comment:
I don't see $faker being created here so that's probably an undefined variable?

Reply:
*facepalm*